shardingsphere-agent/shardingsphere-agent-plugins/shardingsphere-agent-plugin-metrics/shardingsphere-agent-metrics-prometheus/src/main/java/org/apache/shardingsphere/agent/metrics/prometheus/collector/ProxyInfoCollector.java

```diff
@@ -56,10 +56,10 @@ public List<MetricFamilySamples> collect() {
             return Collections.emptyList();
         }
         Optional<GaugeMetricFamily> proxyInfo = FACTORY.createGaugeMetricFamily(MetricIds.PROXY_INFO);
-        Optional<StateContext> stateContext = ProxyContext.getInstance().getStateContext();
-        if (!proxyInfo.isPresent() || !stateContext.isPresent()) {
+        if (null == ProxyContext.getInstance().getContextManager() || !proxyInfo.isPresent()) {
```

Comment:
Why was the judgment of `!stateContext.isPresent()` removed?

Reply:
I thought that when the ContextManager is not empty, the stateContext will definitely not be empty.
According to ProxyContext to obtain the internal logic of stateContext, judgment should be added.